waspc/data/Generator/templates/sdk/wasp/tsconfig.json

```diff
@@ -1,54 +1,106 @@
-{{={= =}=}}
 {
-  "extends": "@tsconfig/node{= majorNodeVersion =}/tsconfig.json",
   "compilerOptions": {
+    /* Basic opitions */
+    "target": "es2022",
     "jsx": "preserve",
     "lib": [
       "esnext",
       "dom",
       "DOM.Iterable"
     ],
-    "declaration": true,
-    "strict": false,
-    "strictBindCallApply": true,
-    "alwaysStrict": true,
-    "noImplicitThis": true,
-    "strictFunctionTypes": true,
-    "strictNullChecks": true,
-    // See https://github.com/wasp-lang/wasp/issues/2056 before activating this:
-    // "useUnknownInCatchVariables": true,
-    // The following 2 stict options will require more work:
-    // "noImplicitAny": true,
-    // "strictPropertyInitialization": true,
-    // Overriding this because we want to use top-level await
-    "module": "esnext",
-    "target": "es2017",
-    // Enable source map for debugging and go-to-definition
-    "sourceMap": true,
-    // The remaining settings should match the extended nodeXY/tsconfig.json, but I kept
-    // them here to be explicit.
-    // Enable default imports in TypeScript.
-    "esModuleInterop": true,
-    "moduleResolution": "node",
-    "outDir": "dist",
+    "skipLibCheck": true,
     "allowJs": true,
+    "rootDir": ".",
+    "outDir": "dist", 
+
+    /* Strictness and type checking */
+    /*
+      User code was compiled uses `"strict": true`, so we know we can use it
+      here as well. We override some of strict's options because parts of the SDK
+      code don't respec them. This only makes the SDK ruleset more permissive than
+      the user's ruleset. Therefore, we can sefely include users code inside the
+      SDK without casuing errors during the SDK build step.
+      Reference: https://github.com/wasp-lang/wasp/issues/1938
+    */
+    "strict": true,
+    "useUnknownInCatchVariables": false, 
+    "noImplicitAny": false,
+    "strictPropertyInitialization": false,
+    // The following two properties are disabled because we don't
+    // yet want to force user code to use them.
+    // Reference: https://github.com/wasp-lang/wasp/issues/2655
+    "noUncheckedIndexedAccess": false,
+    "noImplicitOverride": false,
+
+    /* DX */
+    "declaration": true, // Allows users to consume the SDK as a library, enables autocomplete
+    "declarationMap": true, // Makes go-to-defintion go to source instead of type definitios
+    "sourceMap": true, // Needed for debugging
```

Comment:
`sourceMap` and `declaration` are just moved. `declarationMap` is new and fixes #1986